homeassistant/components/saj/sensor.py

```diff
@@ -0,0 +1,195 @@
+            # and if so: set state to STATE_UNKNOWN.
+            # Sensors with live values like "temperature" or "current_power"
+            # will also be reset to STATE_UNKNOWN.
+            for sensor in hass_sensors:
+                if (sensor.per_day_basis and date.today() > sensor.date_updated) or (
+                    not sensor.per_day_basis and not sensor.per_total_basis
+                ):
+                    task = sensor.async_update_values(unkown_state=True)
+                    if task:
+                        tasks.append(task)
+            if tasks:
+                await asyncio.wait(tasks)
+
+            return
+
+        values = await saj.read(used_sensors)
+        if not values:
+            try:
+                backoff = [1, 1, 1, 6, 30][backoff_step]
+                backoff_step += 1
+            except IndexError:
+                backoff = 60
+            return
+        backoff_step = 0
+
+        for sensor in hass_sensors:
+            task = sensor.async_update_values()
+            if task:
+                tasks.append(task)
+        if tasks:
+            await asyncio.wait(tasks)
+
+    interval = config.get(CONF_SCAN_INTERVAL) or timedelta(seconds=5)
+    async_track_time_interval(hass, async_saj, interval)
+
+
+class SAJsensor(Entity):
+    """Representation of a SAJ sensor."""
+
+    def __init__(self, pysaj_sensor):
+        """Initialize the sensor."""
+        self._sensor = pysaj_sensor
+        self._state = self._sensor.value
+
+    @property
+    def name(self):
+        """Return the name of the sensor."""
+        return f"saj_{self._sensor.name}"
+
+    @property
+    def icon(self):
+        """Return the icon for the sensor."""
+        return SENSOR_ICON
+
+    @property
+    def state(self):
+        """Return the state of the sensor."""
+        return self._state
+
+    @property
+    def unit_of_measurement(self):
+        """Return the unit the value is expressed in."""
+        return self._sensor.unit
+
+    @property
+    def should_poll(self) -> bool:
+        """SAJ sensors are updated & don't poll."""
+        return False
+
+    @property
+    def per_day_basis(self) -> bool:
+        """Return if the sensors value is on daily basis or not."""
+        return self._sensor.per_day_basis
+
+    @property
+    def per_total_basis(self) -> bool:
+        """Return if the sensors value is cummulative or not."""
+        return self._sensor.per_total_basis
+
+    @property
+    def date_updated(self) -> date:
+        """Return the date when the sensor was last updated."""
+        return self._sensor.date
+
+    def async_update_values(self, unkown_state=False):
+        """Update this sensor."""
+        update = False
+
+        if self._sensor.value != self._state:
+            update = True
+            self._state = self._sensor.value
+
+        if unkown_state and self._sensor.value != STATE_UNKNOWN:
```

Comment:
`self._sensor.value` is part of the saj api, right? We shouldn't couple the saj api with the home assistant api directly by using a home assistant constant when checking the saj api.

If an app wide constant changes the integration would break. Please define a constant for the saj api specifically in this module.

Reply:
Woops, that's my fault. It should check for `None` on `self._state` and not `self._sensor.value`.
I'll change it.